File: Integer-to-Roman/Integer-to-Roman.cpp

class Solution {
public:
    string intToRoman(int num) {
        string op[]={"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
        int val[]={1000,900,500,400,100,90,50,40,10,9,5,4,1};
        string ans="";
        for(int i=0;i<13;i++)
        {
            int cnt=num/val[i];
            while(cnt--)
                ans+=op[i];
            num%=val[i];
        }
        return ans;
    }
};


 

File: solutions/Integer-to-Roman/Integer-to-Roman.py


class Solution:
    # @return a string
    def intToRoman(self, num):
        op=["M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"]
        val=[1000,900,500,400,100,90,50,40,10,9,5,4,1]
        ans=""
        for i in range(13):
            cnt=num/val[i]
            while cnt:
                ans+=op[i]
                cnt=cnt-1
            num%=val[i]
        return ans;
 